### PR TITLE
Jjfreund/milestone6 domain take2

### PR DESCRIPTION
**Pull request by jjfreund**
Added models for users (stripped down), reviews, ratings, and keywords

### CODE REVIEW

File: db/migrate/20160626215724_create_reviews.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+class CreateReviews < ActiveRecord::Migration
+  def change
+    create_table :reviews do |t|
+      t.text :review
+      t.references :rating, index: true, foreign_key: true
```
Comment (jfhamlin) [inline db/migrate/20160626215724_create_reviews.rb]: Should this be non-nullable?

File: db/migrate/20160626235137_create_join_table_category_keyword.rb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+class CreateJoinTableCategoryKeyword < ActiveRecord::Migration
+  def change
+    create_join_table :Categories, :Keywords do |t|
```
Comment (jfhamlin) [inline db/migrate/20160626235137_create_join_table_category_keyword.rb]: `ActiveRecord` appears to normalize them, but I think [the parameters are conventionally lowercase](http://apidock.com/rails/ActiveRecord/ConnectionAdapters/SchemaStatements/create_join_table) (as the table names are).

File: db/migrate/20160626214555_create_ratings.rb [hunk 3]
```diff
@@ -0,0 +1,10 @@
+class CreateRatings < ActiveRecord::Migration
+  def change
+    create_table :ratings do |t|
+      t.decimal :rating
+      t.references :user, index: true, foreign_key: true
```
Comment (jfhamlin) [inline db/migrate/20160626214555_create_ratings.rb]: I think this column should be non-null.

File: db/migrate/20160626235137_create_join_table_category_keyword.rb (hunk 2 above)
Comment (jfhamlin) [inline db/migrate/20160626235137_create_join_table_category_keyword.rb]: Ah, `ActiveRecord` doesn't normalize them when creating the join table. These should def be lower case.

File: db/migrate/20160626215946_create_keywords.rb [hunk 4]
```diff
@@ -0,0 +1,7 @@
+class CreateKeywords < ActiveRecord::Migration
+  def change
+    create_table :keywords do |t|
```
Comment (jfhamlin) [inline db/migrate/20160626215946_create_keywords.rb]: If these aren't surfaced in the UI and are just to improve search quality, I'd still hold off on introducing keywords till after a basic search feature is implemented and there's a clear need.

File: db/migrate/20160626214555_create_ratings.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+      t.references :resource, index: true, foreign_key: true
+      t.references :service, index: true, foreign_key: true
```
Comment (jfhamlin) [inline db/migrate/20160626214555_create_ratings.rb]: I'm curious if what we've got planned for the UX for ratings and reviews: what's that interaction going to look like, and are we building both in the front-end immediately?

File: app/models/rating.rb [hunk 6]
```diff
@@ -0,0 +1,6 @@
+class Rating < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :resource
+  belongs_to :service
+  has_one :review
```
Comment (lelandg559) [inline app/models/rating.rb]: should this also has_one :user ?
(the user that made the rating)

Comment (lelandg559) [inline app/models/rating.rb]: this was incorrect, to be clear

belongs_to will place the foreign key in the declaring model whereas has_one will place it in the other model.

we definitely want rating to have a foreign key to user , not user having a foreign key to rating.

**Comment by lelandg559** [comment]
LGTM, can talk about adjusting the keywords when we get to implementing that.

**Pull request merged**